### PR TITLE
feat: add types for event-source-polyfill

### LINKED ISSUE #52957: Definition Request: event-source-polyfill
https://github.com/Yaffle/EventSource

### PR DESCRIPTION
**Pull request by forivall**
Fixes #52957

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/event-source-polyfill/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+// Type definitions for event-source-polyfill 1.0
+// Project: https://github.com/Yaffle/EventSource
+// Definitions by: Emily Klassen <https://github.com/forivall>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.4
```
Comment (peterblazejewicz) [inline types/event-source-polyfill/index.d.ts:5]:
```suggestion
```
you can skip this, 3.7 is default now here

File: types/event-source-polyfill/tslint.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "no-empty-interface": false,
+        "interface-over-type-literal": false
+    }
```
Comment (peterblazejewicz) [inline types/event-source-polyfill/tslint.json:6]:
```suggestion
    "extends": "dtslint/dt.json"
```
please add those inline, as per readme

File: types/event-source-polyfill/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    data: any;
+    lastEventId: string;
+}
+
+export class EventSourcePolyfill {
+    static readonly CLOSED: number;
+    static readonly CONNECTING: number;
+    static readonly OPEN: number;
+
+    constructor(url: string, options: EventSourcePolyfillInit);
```
Comment (peterblazejewicz) [inline types/event-source-polyfill/index.d.ts:52]:
```suggestion
    constructor(url: string, options?: EventSourcePolyfillInit);
```
are options required?

File: types/event-source-polyfill/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    onmessage: ((this: EventSource, ev: MessageEvent) => any) | null;
+    onopen: ((this: EventSource, ev: Event) => any) | null;
+
+    readonly readyState: number;
+    readonly url: string;
+    readonly withCredentials: boolean;
+    close(): void;
+    readonly CLOSED: number;
+    readonly CONNECTING: number;
+    readonly OPEN: number;
```
Comment (peterblazejewicz) [inline types/event-source-polyfill/index.d.ts:64]:
```suggestion
    readonly readyState: number;
    readonly url: string;
    readonly withCredentials: boolean
    readonly CLOSED: number;
    readonly CONNECTING: number;
    readonly OPEN: number;
    
    close(): void;
```

File: types/event-source-polyfill/index.d.ts (hunk 3 above)
Comment (forivall) [reply types/event-source-polyfill/index.d.ts]: my mistake. they're optional.

File: types/event-source-polyfill/index.d.ts (hunk 4 above)
Comment (forivall) [reply types/event-source-polyfill/index.d.ts]: yeah, i used the same ordering that's in `lib.dom.d.ts`. accepted suggestion.

File: types/event-source-polyfill/tslint.json (hunk 2 above)
Comment (forivall) [reply types/event-source-polyfill/tslint.json]: ah, yeah, i was reading react's typings, alas this is noted

> (There are some legacy lint configs that have additional contents, but these should not happen in new work.)

Thanks!

File: types/event-source-polyfill/index.d.ts (hunk 1 above)
Comment (forivall) [reply types/event-source-polyfill/index.d.ts:5]: locally, i'm getting this:
```
» yarn test event-source-polyfill
yarn run v1.22.17
$ dtslint types event-source-polyfill
Error: /Users/forivall/code/repos/DefinitelyTyped/types/event-source-polyfill/event-source-polyfill-tests.ts:20:7
ERROR: 20:7  expect  Compile error in typescript@3.3 but not in typescript@3.4.
Fix with a comment '// Minimum TypeScript Version: 3.4' just under the header.
Type 'EventSourcePolyfill' is not assignable to type 'EventSource'.
  Types of property 'onerror' are incompatible.
    Type '((this: EventSource, ev: Event) => any) | null' is not assignable to type '(evt: MessageEvent) => any'.
      Type 'null' is not assignable to type '(evt: MessageEvent) => any'.

    at /Users/forivall/code/repos/DefinitelyTyped/node_modules/dtslint/bin/index.js:206:19
    at Generator.next (<anonymous>)
    at fulfilled (/Users/forivall/code/repos/DefinitelyTyped/node_modules/dtslint/bin/index.js:6:58)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

i'll remove it and see if the failure happens in CI

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@forivall thx!

**Comment by forivall** [comment]
Ready to merge

**Pull request merged**